ddtrace/context.py

```diff
@@ -125,9 +132,56 @@ def _traceparent(self):
         # type: () -> str
         if self.trace_id is None or self.span_id is None:
             return ""
+        tp = self._meta.get(_TRACEPARENT_KEY)
+        if tp:
+            # grab the original traceparent trace id, not the converted value
+            trace_id = tp.split("-")[1]
+        else:
+            trace_id = "{:032x}".format(self.trace_id)
 
         sampled = 1 if self.sampling_priority and self.sampling_priority > 0 else 0
-        return "00-{:032x}-{:016x}-{:02x}".format(self.trace_id, self.span_id, sampled)
+        return "00-{}-{:016x}-{:02x}".format(trace_id, self.span_id, sampled)
+
+    @property
+    def _tracestate(self):
+        # type: () -> str
+        dd = ""
+        if self.sampling_priority:
+            dd += "s:{};".format(self.sampling_priority)
+        if self.dd_origin:
+            dd += "o:{};".format(self.dd_origin)
+        sampling_decision = self._meta.get(SAMPLING_DECISION_TRACE_TAG_KEY)
+        if sampling_decision:
+            # replace characters ",", "=", and characters outside the ASCII range 0x20 to 0x7E
+            dd += "t.dm:{};".format(re.sub(r",|=|[^\x20-\x7E]+", "_", sampling_decision))
+        # since this can change, we need to grab the value off the current span
+        usr_id_key = self._meta.get(USER_ID_KEY)
+        if usr_id_key:
+            dd += "t.usr.id:{};".format(self._meta.get(USER_ID_KEY))
+
+        # grab all other _dd.p values out of meta since we need to propagate all of them
+        for k, v in self._meta.items():
+            if isinstance(k, str) and k.startswith("_dd.p") and k not in [SAMPLING_DECISION_TRACE_TAG_KEY, USER_ID_KEY]:
```

Comment:
In python2 has two text types: [unicode and str](https://uwpce-pythoncert.github.io/SystemDevelopment/unicode.html#strings-vs-unicode).  So `isinstance(u"some_tag_key", str)` will evaluate to False in python2. 

We can use `isinstance(k, six.string_types)` instead:

```suggestion
            if isinstance(k, six.string_types) and k.startswith("_dd.p") and k not in [SAMPLING_DECISION_TRACE_TAG_KEY, USER_ID_KEY]:
```